src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -257,13 +265,16 @@ class DefaultEntityProvider(
      */
     override fun numEntities(): Int = nextId - recycledEntities.size
 
+    private val currentEntityVersions = Bag(Array(initialEntityCapacity) { 0L })
```

Comment:
`Int` should be sufficient. Entity is also just an `Int`, therefore it doesn't make sense imo to make the version a `Long`.